docs/architecture_decisions.md

```diff
@@ -0,0 +1,62 @@
+## Architecture Decisions
```

Comment:
Can we replace the content with https://github.com/Cloud-CV/EvalAI/wiki/Architectural-Decisions? I have fixed some of the grammatical mistakes in this.